pkg/addonfactory/addonfactory.go

```diff
@@ -47,12 +52,16 @@ func NewAgentAddonFactory(addonName string, fs embed.FS, dir string) *AgentAddon
 			InstallStrategy: nil,
 		},
 		trimCRDDescription: false,
+		scheme:             s,
 	}
 }
 
 // WithScheme is an optional configuration, only used when the agentAddon has customized resource types.
-func (f *AgentAddonFactory) WithScheme(scheme *runtime.Scheme) *AgentAddonFactory {
-	f.scheme = scheme
+func (f *AgentAddonFactory) WithScheme(s *runtime.Scheme) *AgentAddonFactory {
+	f.scheme = s
+	_ = scheme.AddToScheme(f.scheme)
```

Comment:
why twice?

Reply:
NewAgentAddonFactory gives the scheme a default value.
WithScheme also needs to add these schemes since the received s may not haven't these scheme

Reply:
builder := runtime.NewSchemeBuilder(s)
builder.AddToScheme(f.scheme)